dcpy/extract/ingest.py

```diff
@@ -84,10 +90,92 @@ def extract_and_archive_raw_dataset(dataset: str, version: str | None):
     return config
 
 
-def transform_to_parquet(config: recipes.ExtractConfig):
-    """Given config of archived raw dataset, transform it to parquet"""
-    ### idea is that this will dump to PARQUET_PATH - other functions will assume parquet file is there as well
-    raise NotImplemented
+def transform_to_parquet(
+    config: recipes.ExtractConfig, local_data_path: Path | None = None
+):
+    """
+    Transforms raw data into a parquet file format and saves it locally.
+
+    This function first checks for the presence of raw data locally at the specified `local_data_path`.
+    If the path not provided, it is downloaded from an S3 bucket using the `config` parameter.
+    The raw data is then read into a GeoDataFrame and saved as a parquet file.
+
+    The transformation process varies depending on the format of the raw data, which can be in .shp, .gdb,
+    or .csv format. For csv files, if geometry is present, it is converted into a GeoSeries before creating
+    the GeoDataFrame.
+
+    Parameters:
+        config (recipes.ExtractConfig): Config object containing geometry info.
+        local_data_path (Path, optional): Path to the local data file. If not provided, data is pulled from S3 bucket.
+
+    Raises:
+        AssertionError: If `local_data_path` is provided but does not point to a valid file.
+    """
+
+    # create new dir for raw data and output parquet file
+    if TMP_DIR.is_dir():
+        shutil.rmtree(TMP_DIR)
+    TMP_DIR.mkdir()
+
+    if local_data_path:
+        # TODO: fix logic for cases when it's a directory, not a file
+        assert local_data_path.is_file(), "Local path should be a valid file"
+        logger.info(f"Raw data was found locally at {local_data_path}")
+    else:
+        local_data_path = TMP_DIR / config.raw_filename
+
+        s3.download_file(
+            bucket=recipes.BUCKET,
+            key=config.raw_dataset_s3_filepath,
+            path=local_data_path,
+        )
+        logger.info(f"Downloaded raw data from s3 to {local_data_path}")
+
+    data_load_config = config.transform_to_parquet_metadata
+
+    match data_load_config.format:
+        case "shapefile" | "geodabase":
+            gdf = gpd.read_file(
+                local_data_path,
+                geometry=data_load_config.geometry.geom_column,
```

Comment:
Are these fields allowed to be null? For gpd.read_file's signature, I mean. 

For shapefile and geodatabase (typo there btw on line 137), is that a relevant field? Or since it's a special field, can gpd just tell without it being specified?

Reply:
Thanks for catching the misspelling! I realized I never ran a test on "geodabase" format because of it. I will fix it. 

Regarding your questions, I would say no for geospatial data formats because it successfully converts the geom column to wkb format in parquet (which is specific to geom columns): 

<img width="1090" alt="Screenshot 2024-03-19 at 11 30 37 AM" src="https://github.com/NYCPlanning/data-engineering/assets/144725249/8a1fff4c-2270-4424-8b16-ed7e9d33974e">

Having said that, `geom_column` is relevant for cases when we have multiple geometries

Reply:
Do geodatabases and shapefiles support multiple geometries? For a single layer in a geodatabase.

Reply:
Now that I'm thinking about it, they don't support it. It's more relevant to parquet/geoparquet, and text data formats. Should I remove the `geom_column` from the fn signature?